xarray/plot/plot.py

```diff
@@ -429,6 +557,273 @@ def hist(
     return primitive
 
 
+def scatter(
```

Comment:
```suggestion
@plot2d
def scatter(
```

should let you remove some boilerplate code such as facetgrid handling.

Reply:
This is what I tried initially and it was _incredibly_ difficult to fit them together so I gave up on it. I have changed the code quite a bit since then (in particular it's always single pathcollection now instead of possibly list of multple pcs.) so maybe it works now. But I think it's for a later PR, I still feel the burn from my last attempt.